src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -349,14 +349,17 @@ public boolean onKeyAddWord() {
 
 	public boolean onKeyNextLanguage() {
 		if (nextLang()) {
-			commitCurrentSuggestion(false);
 			mInputMode.changeLanguage(mLanguage);
-			mInputMode.reset();
+			mInputMode.clearWordStem();
+			getSuggestions();
 			resetKeyRepeat();
-			clearSuggestions();
+
 			statusBar.setText(mInputMode.toString());
 			mainView.render();
 			forceShowWindowIfHidden();
+			if (!isSuggestionViewHidden()) {
```

Comment:
```suggestion
			if (mInputMode.isABC() && !isSuggestionViewHidden()) {
```

In Predictive mode it will be obvious the language has changed, because the word list would become entirely different. So, let's reduce the toast messages to the bare minimum.